server/home/home-service/src/main/java/io/holoinsight/server/home/biz/plugin/AbstractHostingAlertPlugin.java

```diff
@@ -0,0 +1,183 @@
+    Map<Long, AlarmRuleDTO> ids = new HashMap<>();
+    for (AlarmRuleDTO alarmRuleDTO : alarmRuleDTOList) {
+      Long id = this.alertRuleService.save(alarmRuleDTO);
+      ids.put(id, alarmRuleDTO);
+    }
+    List<AlarmWebhook> webhooks = getWebhookConfig(integrationPluginDTO.tenant);
+    if (!CollectionUtils.isEmpty(webhooks)) {
+      for (AlarmWebhook alarmWebhook : webhooks) {
+        enableAlertSubscribe(alarmWebhook, ids);
+      }
+    }
+  }
+
+  private void enableAlertSubscribe(AlarmWebhook alarmWebhook, Map<Long, AlarmRuleDTO> ruleIds) {
+    if (CollectionUtils.isEmpty(ruleIds)) {
+      return;
+    }
+    for (Map.Entry<Long, AlarmRuleDTO> entry : ruleIds.entrySet()) {
+      AlarmRuleDTO rule = entry.getValue();
+      String uniqueId = rule.getRuleType() + "_" + entry.getKey();
+      AlarmSubscribe alarmSubscribe = new AlarmSubscribe();
+      alarmSubscribe.setCreator(alarmWebhook.getCreator());
+      alarmSubscribe.setGroupId(alarmWebhook.getId());
+      alarmSubscribe.setUniqueId(uniqueId);
+      alarmSubscribe.setNoticeType("[\"webhook\"]");
+      alarmSubscribe.setStatus((byte) 1);
+      alarmSubscribe.setTenant(alarmWebhook.getTenant());
+      alarmSubscribe.setEnvType(rule.getEnvType());
+      this.alarmSubscribeMapper.insert(alarmSubscribe);
+    }
+  }
+
+  private List<AlarmWebhook> getWebhookConfig(String tenant) {
+    QueryWrapper<AlarmWebhook> queryWrapper = new QueryWrapper<>();
+    queryWrapper.eq("role", "hosting");
+    queryWrapper.eq("tenant", tenant);
+    return this.alarmWebhookMapper.selectList(queryWrapper);
+  }
+
+  private List<Filter> buildFilter(IntegrationPluginDTO newPlugin) {
+    Map<String, Object> collectRange = newPlugin.getCollectRange();
+    if (CollectionUtils.isEmpty(collectRange)) {
+      return Collections.emptyList();
+    }
+    IntegrationPluginDTO.DataRange dataRange =
+        J.fromJson(J.toJson(collectRange), IntegrationPluginDTO.DataRange.class);
+    if (CollectionUtils.isEmpty(dataRange.getValuesMap())) {
+      return Collections.emptyList();
+    }
+    List<Filter> list = new ArrayList<>();
+
+    for (Map.Entry<String, String> entry : dataRange.getValuesMap().entrySet()) {
+      String key = entry.getKey();
+      Filter filter = new Filter();
+      filter.setName(key);
+      filter.setValue(entry.getValue());
+      filter.setType("literal_or");
+      list.add(filter);
+    }
+    return list;
+  }
+
+  protected boolean disableAlertHosting(IntegrationPluginDTO disableIntegrationPlugin) {
+    Long id = disableIntegrationPlugin.id;
+    if (id == null) {
+      return true;
+    }
+
+    AlarmRuleDTO condition = new AlarmRuleDTO();
+    condition.setSourceId(id);
+    MonitorPageRequest<AlarmRuleDTO> request = new MonitorPageRequest<>();
+    request.setTarget(condition);
+    request.setPageSize(1000);
+    MonitorPageResult<AlarmRuleDTO> result = this.alertRuleService.getListByPage(request);
```

Comment:
use listAll() instead，has a problem when using pageRequest

Reply:
Get it.